src/librustc_resolve/diagnostics.rs

```diff
@@ -870,7 +877,9 @@ impl<'a> Resolver<'a> {
                     // otherwise cause duplicate suggestions.
                     continue;
                 }
-                if let Some(crate_id) = self.crate_loader.maybe_process_path_extern(ident.name) {
+                if let Some(crate_id) =
+                    self.crate_loader.maybe_process_path_extern(ident.name, self.avoid_crate_load)
```

Comment:
This PR should not modify anything in `rustc_metadata`.

If `find_unique_symbols` runs during any execution and not only when errors are guaranteed to happen, then it should either
- not rely on `lookup_import_candidates` because it loops through the whole `self.extern_prelude`.
- loop only through crates that are already loaded, using `cstore.crates_untracked()` or other function listing the loaded `CrateNum`s.

Reply:
Based on this, found a way to modify `lookup_import_candidates` so that `rustc_metadata` can stay the same. Done in a subsequent commit.